features/facebookImport/src/model/analyses/ministories/about-pictures-data-analysis.js

```diff
@@ -44,7 +44,7 @@ export default class AboutPicturesDataAnalysis extends RootAnalysis {
     async analyze({ zipFile, dataAccount }) {
         const pictureEntries = await this._userPicturesFromExport(zipFile);
         if (pictureEntries.length >= PICTURES_THRESHOLD)
-            dataAccount.analyses[analysisKeys.picturesCount] =
+            dataAccount.processedData[analysisKeys.picturesCount] =
```

Comment:
This is again violating encapsulation. Option 3 above would turn this into an "analyzer" that would be injected in the object at construction, and triggered when data becomes available.
That's another problem we have with the option we're seeing. Since data mutation is not handled by the object, we don't know if data is consistent. We hope it is, and specifically all data that's hold at one specific point in time belongs to the same source. But since this is not checked, we could be in a situation in which we're reusing the object for a new import and the imported files are not consistent with imported data. You can argue "we are never going to do that", but the things that we're never gonna do have to be enforced by the data model.
Additionally, this simply supposes that "processed data" is going to be an object that has certain strings as keys. If essentially processed data is going to be handled on the client side as key-value pairs, this could be turned into an "Add processed data" method with encapsulated storage (could be later a triple store, we don't know at this point) and additional business logic (checking for duplicated keys, for instance).

Reply:
Returning a reference via getter to a private attribute, at the end of the day, makes it public. Only difference is that you can't assign it to a string or whatever, but still.